### ISSUE TITLE
Numpy issue on o2

### ISSUE THREAD

**Issue by DenisSch** [labsyspharm/mcmicro#32]
```
Error executing process > 'unmicst (3)'

Caused by:
  Process `unmicst (3)` terminated with an error exit status (1)

Command executed:

  python /n/groups/lsp/mcmicro/UnMicst/UnMicst.py 4.tif --outputPath .

Command exit status:
  1

Command output:
  (empty)

Command error:
  Traceback (most recent call last):
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/__init__.py", line 40, in <module>
      from . import multiarray
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/multiarray.py", line 13, in <module>

 File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/overrides.py", line 6, in <module>
      from numpy.core._multiarray_umath import (
  ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "/n/groups/lsp/mcmicro/UnMicst/UnMicst.py", line 1, in <module>
      import numpy as np
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/__init__.py", line 142, in <module>
      from . import core
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/__init__.py", line 71, in <module>
      raise ImportError(msg)
  ImportError:

  IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

  Importing the multiarray numpy extension module failed.  Most
  likely you are trying to import a failed build of numpy.
  Here is how to proceed:
  - If you're working with a numpy git repository, try `git clean -xdf`
    (removes all files not under version control) and rebuild numpy.
  - If you are simply trying to use the numpy version that you have installed:
    your installation is broken - please reinstall numpy.
  - If you have already reinstalled and that did not fix the problem, then:
    1. Check that you are using the Python you expect (you're using /home/ds230/work/conda/unmicst-fd23057379ebdf6804bf723a9b5c3328/bin/python),
       and that you have no directories in your PATH or PYTHONPATH that can
       interfere with the Python and numpy versions you're trying to use.
    2. If (1) looks fine, you can open a new issue at
       https://github.com/numpy/numpy/issues.  Please include details on:
       - how you installed Python
       - how you installed numpy
       - your operating system
       - whether or not you have multiple versions of Python installed
       - if you built from source, your compiler versions and ideally a build log

       Note: this error has many possible causes, so please don't comment on
       an existing issue about this - open a new one instead.

 Original error was: No module named 'numpy.core._multiarray_umath'

Work dir:
  /home/ds230/work/e3/6ac6d6c8315a35e488f44b1f927afa

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

executor >  slurm (5)
[-        ] process > illumination -
[-        ] process > ashlar	   -
[bb/511c76] process > dearray	   [100%] 1 of 1 ✔
[16/e1333b] process > unmicst	   [400%] 4 of 1, failed: 4 ✘
[-        ] process > s3seg        -
WARN: Killing pending tasks (3)
Error executing process > 'unmicst (3)'

aused by:
  Process `unmicst (3)` terminated with an error exit status (1)

Command executed:

  python /n/groups/lsp/mcmicro/UnMicst/UnMicst.py 4.tif --outputPath .

Command exit status:
  1

Command output:
  (empty)

Command error:
  Traceback (most recent call last):
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/__init__.py", line 40, in <module>
      from . import multiarray
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/multiarray.py", line 13, in <module>
      from . import overrides
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/overrides.py", line 6, in <module>
      from numpy.core._multiarray_umath import (
  ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "/n/groups/lsp/mcmicro/UnMicst/UnMicst.py", line 1, in <module>
      import numpy as np
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/__init__.py", line 142, in <module>
      from . import core
    File "/n/app/python/3.7.4-ext/lib/python3.7/site-packages/numpy/core/__init__.py", line 71, in <module>
      raise ImportError(msg)
  ImportError:

  IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

  Importing the multiarray numpy extension module failed.  Most
  likely you are trying to import a failed build of numpy.
  Here is how to proceed:
  - If you're working with a numpy git repository, try `git clean -xdf`
    (removes all files not under version control) and rebuild numpy.
  - If you are simply trying to use the numpy version that you have installed:
    your installation is broken - please reinstall numpy.
  - If you have already reinstalled and that did not fix the problem, then:
    1. Check that you are using the Python you expect (you're using /home/ds230/work/conda/unmicst-fd23057379ebdf6804bf723a9b5c3328/bin/python),
       and that you have no directories in your PATH or PYTHONPATH that can
       interfere with the Python and numpy versions you're trying to use.
    2. If (1) looks fine, you can open a new issue at
       https://github.com/numpy/numpy/issues.  Please include details on:
       - how you installed Python
       - how you installed numpy
       - your operating system
       - whether or not you have multiple versions of Python installed
       - if you built from source, your compiler versions and ideally a build log

       Note: this error has many possible causes, so please don't comment on
       an existing issue about this - open a new one instead.

  Original error was: No module named 'numpy.core._multiarray_umath'

Work dir:
  /home/ds230/work/e3/6ac6d6c8315a35e488f44b1f927afa

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`
```

**Comment by DenisSch** [comment]
I tried to add numpy to the yml file, but it did not change the error message.
@ArtemSokolov Can you run the full pipeline without issues on o2 currently?

**Comment by ArtemSokolov** [comment]
Confirming that I can reproduce the error. Going to compare versions of python packages between Conda and Docker (which runs without issues).

**Comment by ArtemSokolov** [comment]
The issue is due to a conflict in the `numpy` version created by `module load ashlar` and `module load conda2`. Jeremy is working on a conda environment file for ASHLAR to be included with the pipeline (thus removing the need for `module load ashlar`).

As an intermediate solution, ASHLAR and everything post-ASHLAR can be run in two different O2 sessions:
1. `module load gcc ashlar java matlab`
2. Run the pipeline as usual. It will complete stitching (and optionally de-arraying) but fail at unmicst.
3. Start a fresh O2 session.
4. `module load conda2 java`
5. Run the pipeline again but with `--skip_ashlar` flag to use the stitched image generated in Step 2.

**Comment by DenisSch** [comment]
Can you run the full pipeline until the end now?
I am getting the follow error message and I am not sure if related:
```

Error executing process > 's3seg (1)'

Caused by:
  Process `s3seg (1)` terminated with an error exit status (1)

Command executed:

  python /n/groups/lsp/mcmicro/S3segmenter/S3segmenter.py --crop noCrop        --imagePath exemplar-002.ome.tif        --maskPath NO_MASK        --nucleiClassProbPath exemplar-002_NucleiPM_1.tif        --contoursClassProbPath exemplar-002_ContoursPM_1.tif        --outputPath .

Command exit status:
  1

Command output:
  (empty)

Command error:
  Traceback (most recent call last):
    File "/n/groups/lsp/mcmicro/S3segmenter/S3segmenter.py", line 249, in <module>
      tissue_gauss1[tissue_gauss>np.percentile(tissue_gauss,99)]=np.nan
  ValueError: cannot convert float NaN to integer

Work dir:
  /home/ds230/work/00/db65a7a15be0c1828274e12e14cbe2

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`
```

**Comment by DenisSch** [comment]
Correction 4:
```module load conda2 java matlab```

**Comment by ArtemSokolov** [comment]
Issue fixed by #37 and #38.
